cmd/buildah/bud.go

```diff
@@ -309,75 +315,120 @@ func budCmd(c *cobra.Command, inputArgs []string, iopts budOptions) error {
+		func() {
+			platform := platform
+			builds.Go(func() error {
+				platformContext := *systemContext
+				systemContext = &platformContext
+				systemContext.OSChoice = platform.OS
+				systemContext.ArchitectureChoice = platform.Arch
+				systemContext.VariantChoice = platform.Variant
+				options := define.BuildOptions{
+					AddCapabilities:         iopts.CapAdd,
+					AdditionalTags:          tags,
+					Annotations:             iopts.Annotation,
+					Architecture:            platform.Arch,
+					Args:                    args,
+					BlobDirectory:           iopts.BlobCache,
+					CNIConfigDir:            iopts.CNIConfigDir,
+					CNIPluginPath:           iopts.CNIPlugInPath,
+					CommonBuildOpts:         commonOpts,
+					Compression:             compression,
+					ConfigureNetwork:        networkPolicy,
+					ContextDirectory:        contextDir,
+					DefaultMountsFilePath:   globalFlagResults.DefaultMountsFile,
+					Devices:                 iopts.Devices,
+					DropCapabilities:        iopts.CapDrop,
+					Err:                     stderr,
+					ForceRmIntermediateCtrs: iopts.ForceRm,
+					From:                    iopts.From,
+					IDMappingOptions:        idmappingOptions,
+					IIDFile:                 iopts.Iidfile,
+					In:                      stdin,
+					Isolation:               isolation,
+					Labels:                  iopts.Label,
+					Layers:                  layers,
+					LogRusage:               iopts.LogRusage,
+					Manifest:                iopts.Manifest,
+					MaxPullPushRetries:      maxPullPushRetries,
+					NamespaceOptions:        namespaceOptions,
+					NoCache:                 iopts.NoCache,
+					OS:                      platform.OS,
+					Out:                     stdout,
+					Output:                  output,
+					OutputFormat:            format,
+					PullPolicy:              pullPolicy,
+					PullPushRetryDelay:      pullPushRetryDelay,
+					Quiet:                   iopts.Quiet,
+					RemoveIntermediateCtrs:  iopts.Rm,
+					ReportWriter:            reporter,
+					Runtime:                 iopts.Runtime,
+					RuntimeArgs:             runtimeFlags,
+					RusageLogFile:           iopts.RusageLogFile,
+					SignBy:                  iopts.SignBy,
+					SignaturePolicyPath:     iopts.SignaturePolicy,
+					Squash:                  iopts.Squash,
+					SystemContext:           systemContext,
+					Target:                  iopts.Target,
+					TransientMounts:         iopts.Volumes,
+					OciDecryptConfig:        decConfig,
+					Jobs:                    &iopts.Jobs,
+					JobSemaphore:            jobSemaphore,
+				}
+				if iopts.IgnoreFile != "" {
+					excludes, err := parseDockerignore(iopts.IgnoreFile)
+					if err != nil {
+						return err
+					}
+					options.Excludes = excludes
+				}
+				if c.Flag("timestamp").Changed {
+					timestamp := time.Unix(iopts.Timestamp, 0).UTC()
+					options.Timestamp = &timestamp
+				}
+				if iopts.Quiet {
+					options.ReportWriter = ioutil.Discard
+				}
```

Comment:
This also might be possible to do only once, OTOH that would require a larger restructuring.

Reply:
Moving the initializer for `timestamp` out of the loop.